src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1366,11 +1366,39 @@ MavlinkReceiver::handle_message_set_attitude_target(mavlink_message_t *msg)
 						att_sp.yaw_sp_move_rate = 0.0f;
 					}
 
-					// TODO: We assume offboard is only used for multicopters which produce thrust along the
-					// body z axis. If we want to support fixed wing as well we need to handle it differently here, e.g.
-					// in that case we should assign att_sp.thrust_body[0]
 					if (!_offboard_control_mode.ignore_thrust) { // dont't overwrite thrust if it's invalid
-						att_sp.thrust_body[2] = -set_attitude_target.thrust;
+						// Fill correct field by checking frametype
+						// TODO: add as needed
+						switch (_mavlink->get_system_type()) {
+						case MAV_TYPE_GENERIC:
+							break;
+
+						case MAV_TYPE_FIXED_WING:
+							break;
+
+						case MAV_TYPE_QUADROTOR:
+							att_sp.thrust_body[2] = -set_attitude_target.thrust;
```

Comment:
Do you know why the quadcopter case has this `-`?

Reply:
> 
> 
> Do you know why the quadcopter case has this `-`?

Sorry I don 't know ether. The controller of multi-copter is really a complex thing so I just treat it as black-box and just leave it there.

Reply:
Positive z is down.